pandas/io/html.py

```diff
@@ -590,6 +697,12 @@ def _parse_tables(self, doc, match, kwargs):
                              .format(patt=pattern))
         return tables
 
+    def _equals_tag(self, obj, tag):
```

Comment:
aren't these the same as in the super class?

Reply:
I thought this too on initial glance but the attribute of `obj` used to determine equality differs across implementations. An alternate approach could be to make that attribute a property that the subclasses need to implement, though not sure that is that much cleaner

Reply:
My answer is either "no" or "I don't understand the question" :).

But I did notice and nix `_contains_tag()`, which did not belong. So thanks for focusing here :).

Reply:
The bizarre inheritance used in this module makes "node-like" values different across implementations. It isn't duck-typing: they're completely different beasts. That's what it was like before I started this pull request.

Reply:
ok happy to take follow-ups to make this simpler / easier for implementation.